bson/src/main/org/bson/codecs/pojo/ConventionObjectIdGeneratorsImpl.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.bson.codecs.pojo;
+
+import org.bson.BsonObjectId;
+import org.bson.types.ObjectId;
+
+final class ConventionObjectIdGeneratorsImpl implements Convention {
+    @Override
+    public void apply(final ClassModelBuilder<?> classModelBuilder) {
```

Comment:
What would happen if there was an embedded Pojo with an id property?  Would this make the driver to generate an id for those instances as well?  This is probably the wrong place in the code to ask that question, but I'm not sure where to look.

Reply:
It turned out not to be an issue with the initial approach, but once I moved the logic into the PojoCodecImpl you could see _id's being generated at each level.  The LevelCountingBsonWriter fixed that issue.